File: reverse-a-doubly-linked-list.java
public static Node reverseDLL(Node head)
{
    //Your code here
    Node current = head;
    Node temp = null;

    // Swap next and prev pointers for all nodes of the doubly linked list
    while (current != null) {
        temp = current.prev;
        current.prev = current.next;
        current.next = temp;
        current = current.prev;
    }

    // Update the head
    if (temp != null) {
        head = temp.prev;
    }

    return head;
}

File: Difficulty: Easy/Reverse a Doubly Linked List/reverse-a-doubly-linked-list.cpp
class Solution {
  public:
    Node *reverse(Node *head) {
        // code here
        Node* curr = head;
        int l = 0 , r = 0;
        while(curr -> next != NULL){
            curr = curr -> next;
            r++;
        }
        
        Node* prevNode = curr;
        curr = head;
        while(l <= r){
            int temp = curr -> data;
            curr -> data = prevNode -> data;
            prevNode -> data = temp;
            curr = curr -> next;
            prevNode = prevNode -> prev;
            l++;
            r--;
        }
        return head;
    }
};